crates/miden-objects/src/note/note_tag.rs

```diff
@@ -195,27 +205,39 @@ impl NoteTag {
     ///
     /// A note is intended for execution by a single account if the first two bits are zeros
     pub fn is_single_target(&self) -> bool {
-        let first_2_bit = self.0 >> 30;
-        first_2_bit == 0b00
+        match self {
+            NoteTag::NetworkAccount(_) | NoteTag::LocalAccount(_) => true,
+            NoteTag::NetworkUseCase(_, _) | NoteTag::LocalUseCase(_, _) => false,
+        }
     }
 
     /// Returns note execution mode defined by this tag.
     ///
     /// If the most significant bit of the tag is 0 the note is intended for local execution;
     /// otherwise, the note is intended for network execution.
     pub fn execution_mode(&self) -> NoteExecutionMode {
-        let first_bit = self.0 >> 31;
-
-        if first_bit == (LOCAL_EXECUTION as u32) {
-            NoteExecutionMode::Local
-        } else {
-            NoteExecutionMode::Network
+        match self {
+            NoteTag::NetworkAccount(_) | NoteTag::NetworkUseCase(_, _) => {
+                NoteExecutionMode::Network
+            },
+            NoteTag::LocalAccount(_) | NoteTag::LocalUseCase(_, _) => NoteExecutionMode::Local,
         }
     }
 
     /// Returns the inner u32 value of this tag.
     pub fn inner(&self) -> u32 {
```

Comment:
I think we should rename this to `as_u32`, because it's much clearer when reading it elsewhere.